components/MechAgents.tsx

```diff
@@ -0,0 +1,191 @@
+import { useQuery } from '@tanstack/react-query';
+import { readContract } from '@wagmi/core';
+import { Flex, Table, Typography } from 'antd';
+import { getMechAgents } from 'graphql/queries';
+import { MechAgent } from 'graphql/types';
+import Jazzicon, { jsNumberForAddress } from 'react-jazzicon';
+import styled from 'styled-components';
+import { UsePublicClientReturnType, usePublicClient } from 'wagmi';
+
+import { Card } from 'components/shared/styles';
+import { AGENT_REGISTRY_ABI, AGENT_REGISTRY_ADDRESS } from 'constants/contracts/agentRegistry';
+import { GNOSIS_SCAN_URL, NA } from 'constants/index';
+import { COLOR } from 'constants/theme';
+import { wagmiConfig } from 'constants/wagmiConfig';
+import { getAgentName } from 'utils/agents';
+import { getIpfsResponse } from 'utils/ipfs';
+
+const { Title, Text, Paragraph } = Typography;
+
+const Tag = styled.div`
+  display: flex;
+  padding: 2px 6px;
+  border-radius: 6px;
+  background: ${COLOR.WHITE_TRANSPARENT_5};
+  color: ${COLOR.SECONDARY};
+`;
+
+const LONG_HASHES_AGENT_ID = 6;
+
+// TODO: it's impossible to use multicall for all agents, because for some agents
+// the result is too long and the request fails. Ideally need to move it
+// to the subgraph; using this approach to save time
```

Comment:
to add to this comment: we already have a subgraph for the mech agents https://github.com/valory-xyz/ai-registry-mech-graph and ideally I would've extended this subgraph with the agent registry contract, but the subgraph is in use already at least here https://aimechs.autonolas.network/registry (btw, this is why the page is unstable - because for 6th agent the getHashes returns 4800 hashes) and maybe somewhere else, plus it's deployed on Viraj's studio so I won't be able to re-deploy it there. I used this approach in the interest of time